lib/ts/utils/dateProvider/index.ts

```diff
@@ -0,0 +1,57 @@
+/* Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { defaultDateProviderImplementation } from "./defaultImplementation";
+import { DateProviderInterface, DateProviderInput } from "./types";
+
+/**
+ * A utility class for managing a reference to a DateProviderInterface, allowing customization of time-related operations.
+ * This class is designed to address clock skew issues between the server and client by providing a consistent mechanism
+ * for obtaining current time with the serverClockDeviation.
+ *
+ * @class DateProviderReference
+ */
+export default class DateProviderReference {
+    private static instance?: DateProviderReference;
+
+    dateProvider: DateProviderInterface;
+
+    constructor(dateProviderInput?: DateProviderInput) {
+        let dateProviderFunc: DateProviderInput = original => original;
+        if (dateProviderInput !== undefined) {
+            dateProviderFunc = dateProviderInput;
+        }
+
+        this.dateProvider = dateProviderFunc(defaultDateProviderImplementation);
+    }
+
+    static init(dateProviderInput?: DateProviderInput): void {
```

Comment:
Where is this called? Something seems to be missing here.

Reply:
Thanks for catching that. I have fixed it.

My manual tests passed because it was getting initialized by supertokens-web-js.